File: packages/sdks/client-sdks/gatrix-flutter-sdk/lib/src/variant_source.dart
/// Well-known variant source names shared across all Gatrix SDKs.
class VariantSource {
  VariantSource._();

  /// Flag not found in SDK cache
  static const String missing = r'$missing';

  /// SDK detected a type mismatch between requested and actual value type
  static const String typeMismatch = r'$type-mismatch';

  /// Value from environment-level enabledValue
  static const String envDefaultEnabled = r'$env-default-enabled';

  /// Value from flag-level (global) enabledValue
  static const String flagDefaultEnabled = r'$flag-default-enabled';

  /// Value from environment-level disabledValue
  static const String envDefaultDisabled = r'$env-default-disabled';

  /// Value from flag-level (global) disabledValue
  static const String flagDefaultDisabled = r'$flag-default-disabled';
}

File: packages/sdks/client-sdks/gatrix-unity-client-sdk/src/Assets/Plugins/Gatrix/Runtime/VariantSource.cs
// Reserved variant source names used throughout the SDK.
// These constants define how the origin of a variant value
// is classified when returned from evaluation.

namespace Gatrix.Unity.SDK
{
    /// <summary>
    /// Well-known variant source names shared across all Gatrix SDKs.
    /// </summary>
    public static class VariantSource
    {
        /// <summary>Flag not found in SDK cache</summary>
        public const string Missing = "$missing";

        /// <summary>SDK detected a type mismatch between requested and actual value type</summary>
        public const string TypeMismatch = "$type-mismatch";

        /// <summary>Value from environment-level enabledValue</summary>
        public const string EnvDefaultEnabled = "$env-default-enabled";

        /// <summary>Value from flag-level (global) enabledValue</summary>
        public const string FlagDefaultEnabled = "$flag-default-enabled";

        /// <summary>Value from environment-level disabledValue</summary>
        public const string EnvDefaultDisabled = "$env-default-disabled";

        /// <summary>Value from flag-level (global) disabledValue</summary>
        public const string FlagDefaultDisabled = "$flag-default-disabled";
    }
}
